### PR TITLE
remove some gcc warnings

### PR DESCRIPTION
**Pull request by rwst**
This pull request addresses issue #...

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/trie.c [hunk 1]
```diff
@@ -507,7 +507,7 @@ _iterate_helper(const Trie* trie,
 	const char *suffix = transition->suffix;
 	unsigned int keylen = strlen(current_key);
 
-	if(keylen + strlen(suffix) >= max_key) {
```
[the remark this answers is not in the document]
Comment (rwst) [reply Bio/trie.c]: Right. The parameter gets an unsigned value, anyway.

**Comment by rwst** [comment]
> @mdehoon ought to recorded

In this PR?

**Comment by peterjc** [comment]
I meant this was why GitHub asked me to review, when Michiel would have been my first choice.

I'll fix https://github.com/biopython/biopython/blob/master/.github/CODEOWNERS now...

**Comment by peterjc** [comment]
(Sorry, clicked on the wrong button and closed this by mistake, was meant to be comment only)

**Comment by peterjc** [comment]
See 0327ec6a39aff3eb66689a2069050915c88e4482

**Comment by peterjc** [comment]
The update to ``Bio/trie.c`` looks sensible to me now.

**Comment by peterjc** [comment]
Thank you both - and Ralf, just let us know if you change your mind about being listed as a contributor.

**Comment by rwst** [comment]
I'll certainly want being listed when there is more content.

**Pull request merged**